core/domain/topic_fetchers.py

```diff
@@ -241,28 +267,30 @@ def get_topic_by_url_fragment(url_fragment):
         Topic or None. The domain object representing a topic with the
         given id, or None if it does not exist.
     """
-    topic_model = (
+    topic_model: Optional[topic_models.TopicModel] = (
         topic_models.TopicModel.get_by_url_fragment(url_fragment))
     if topic_model is None:
         return None
 
-    topic = get_topic_from_model(topic_model)
+    topic: topic_domain.Topic = get_topic_from_model(topic_model)
     return topic
 
 
-def get_all_topics():
+def get_all_topics() -> List[topic_domain.Topic]:
     """Returns all the topics present in the datastore.
 
     Returns:
         list(Topic). The list of topics present in the datastore.
     """
     backend_topic_models = topic_models.TopicModel.get_all()
-    topics = [
+    topics: List[topic_domain.Topic] = [
         get_topic_from_model(topic) for topic in backend_topic_models]
     return topics
 
 
-def get_topic_rights(topic_id, strict=True):
+def get_topic_rights(
+    topic_id: str, strict: bool = True
+) -> Optional[topic_domain.TopicRights]:
```

Comment:
Can we add overloads and change the return type conditionally on the `strict` argument?

Reply:
sure. I tried to add overloads, PTAL.